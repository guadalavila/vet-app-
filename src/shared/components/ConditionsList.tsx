import React, { useContext } from 'react';
import { StyleSheet, View } from 'react-native';
import Title from './Title';
import Badge from './Badge';
import { getRandomColor } from '../utils/helpers';
import { size } from '../utils/size';
import { PathologiesContext } from '../../contexts/PathologiesContext';

interface IConditionsListProps {
    conditions: string[];
}

const ConditionsList = ({ conditions }: IConditionsListProps) => {
    const { pathologies } = useContext(PathologiesContext);

    const getConditionColorCode = (name: string) => {
        return pathologies.find((x) => name === x.name)?.colorCode ?? getRandomColor();
    };
    return (
        <>
            <Title text='PatologÃ­a/s preexistentes:' />
            <View style={styles.containerConditions}>
                {conditions.map((item) => (
                    <Badge key={item} label={item} color={getConditionColorCode(item)} />
                ))}
            </View>
        </>
    );
};

export default ConditionsList;

const styles = StyleSheet.create({
    containerConditions: {
        justifyContent: 'flex-start',
        flexDirection: 'row',
        flexWrap: 'wrap',
        paddingHorizontal: size.L,
    },
});
