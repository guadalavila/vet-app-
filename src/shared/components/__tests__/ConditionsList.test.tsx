import { render, screen } from '@testing-library/react-native';
import React from 'react';
import ConditionsList from '../ConditionsList';

describe('ConditionList', () => {
    test('should match snapshot', () => {
        const container = render(<ConditionsList conditions={['Ciego', 'Sordo']} />);
        expect(container).toMatchSnapshot();
    });
    test('should render component', () => {
        render(<ConditionsList conditions={['Ciego', 'Sordo']} />);
        expect(screen.getByText('Patolog√≠a/s preexistentes:')).toBeTruthy();
        expect(screen.getByText('Ciego')).toBeTruthy();
        expect(screen.getByText('Sordo')).toBeTruthy();
        // screen.debug();
        expect(screen.getByTestId('Ciego').props.style).toEqual([
            { color: '#11100F', fontSize: 16 },
            { color: '#11100F', fontWeight: '600' },
        ]);
        expect(screen.getByTestId('Sordo').props.style).toEqual([
            { color: '#11100F', fontSize: 16 },
            { color: '#11100F', fontWeight: '600' },
        ]);
    });
});
