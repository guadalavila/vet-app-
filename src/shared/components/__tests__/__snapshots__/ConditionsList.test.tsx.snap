// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConditionList should match snapshot 1`] = `
Array [
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "600",
            "marginLeft": 14,
            "marginVertical": 14,
          },
        ],
      ]
    }
  >
    Patolog√≠a/s preexistentes:
  </Text>,
  <View
    style={
      Object {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "flex-start",
        "paddingHorizontal": 8,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "borderRadius": 12,
            "margin": 6,
            "padding": 6,
          },
          Object {
            "backgroundColor": "#79a30",
          },
        ]
      }
      testID="container-text"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#11100F",
              "fontSize": 16,
            },
            Object {
              "color": "#11100F",
              "fontWeight": "600",
            },
          ]
        }
        testID="Ciego"
      >
        Ciego
      </Text>
    </View>
    <View
      style={
        Array [
          Object {
            "borderRadius": 12,
            "margin": 6,
            "padding": 6,
          },
          Object {
            "backgroundColor": "#72212d",
          },
        ]
      }
      testID="container-text"
    >
      <Text
        style={
          Array [
            Object {
              "color": "#11100F",
              "fontSize": 16,
            },
            Object {
              "color": "#11100F",
              "fontWeight": "600",
            },
          ]
        }
        testID="Sordo"
      >
        Sordo
      </Text>
    </View>
  </View>,
]
`;
